`docker build` fails

I forgot to update the Dockerfile in PR #124, so now CI fails with the below error ([example](https://github.com/atlanticwave-sdx/sdx-lc/actions/runs/9228921130/job/25395083937)).

```
2024-05-24T19:58:13.3884221Z #9 DONE 0.0s
2024-05-24T19:58:13.3885002Z ------
2024-05-24T19:58:13.3885778Z  > [4/6] COPY requirements.txt /usr/src/app/:
2024-05-24T19:58:13.3886896Z ------
2024-05-24T19:58:13.3887504Z Dockerfile:6
2024-05-24T19:58:13.3888080Z --------------------
2024-05-24T19:58:13.3888777Z    4 |     WORKDIR /usr/src/app
2024-05-24T19:58:13.3889322Z    5 |     
2024-05-24T19:58:13.3889882Z    6 | >>> COPY requirements.txt /usr/src/app/
2024-05-24T19:58:13.3890673Z    7 |     
2024-05-24T19:58:13.3891442Z    8 |     RUN pip3 install --no-cache-dir -r requirements.txt
2024-05-24T19:58:13.3892367Z --------------------
2024-05-24T19:58:13.3893896Z ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref kekyw9au1wan7df33z5l213t6::j9s12384apujgsmmhiiwo23vi: "/requirements.txt": not found
2024-05-24T19:58:13.3963303Z ##[error]buildx failed with: ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref kekyw9au1wan7df33z5l213t6::j9s12384apujgsmmhiiwo23vi: "/requirements.txt": not found
2024-05-24T19:58:13.4089212Z Post job cleanup.
```

There are also warnings about deprecated CI steps.

--- a/.github/workflows/checks.yml
+++ b/.github/workflows/checks.yml
@@ -18,10 +18,10 @@ jobs:
     steps:
 
       - name: Check out sources
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Set up Python 3.9
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
         with:
           python-version: "3.9"
           cache: 'pip'  # cache pip dependencies

--- a/.github/workflows/images.yml
+++ b/.github/workflows/images.yml
@@ -22,13 +22,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       # Collect metadata for container image tags
       # See https://github.com/docker/metadata-action
       - name: Docker meta
         id: meta
-        uses: docker/metadata-action@v4
+        uses: docker/metadata-action@v5
         with:
           # list of Docker images to use as base name for tags
           images: |
@@ -50,7 +50,7 @@ jobs:
       #
       # # See https://github.com/docker/login-action.
       - name: Login to Docker Hub
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
@@ -60,11 +60,11 @@ jobs:
       #
       # See https://github.com/docker/setup-buildx-action.
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+        uses: docker/setup-buildx-action@v3
 
       # See https://github.com/docker/build-push-action
       - name: Build and push
-        uses: docker/build-push-action@v4
+        uses: docker/build-push-action@v5
         with:
           context: .
           file: ./Dockerfile

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -59,7 +59,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install flake8 tox
+          python -m pip install flake8 tox coverage
 
       - name: Lint with flake8
         run: |

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -59,7 +59,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install flake8 tox coveralls
+          python -m pip install flake8 tox
 
       - name: Lint with flake8
         run: |

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -72,7 +72,10 @@ jobs:
 
       - name: Run tests
         run: |
+          # Run tests and collect coverage data.
           python -m tox
+          # Generate LCOV format coverage data for coveralls.
+          python -m coverage lcov -o coverage.lcov
         env:
           SDXLC_HOST: 'localhost'
           SDXLC_DOMAIN: 'example.net'
@@ -93,22 +96,19 @@ jobs:
           MONGODB_CONNSTRING: 'mongodb://guest:guest@localhost:27017/'
 
       - name: Send coverage data to coveralls.io
-        run: |
-          python -m coveralls --service=github
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          COVERALLS_FLAG_NAME: ${{ matrix.python-version }}
-          COVERALLS_PARALLEL: true
+        uses: coverallsapp/github-action@v2
+        with:
+          flag-name: run-${{ join(matrix.*, '-') }}
+          file: coverage.lcov
+          parallel: true
 
   finalize:
     name: finalize
     needs: test
     runs-on: ubuntu-latest
-    container: python:3-slim
+    if: ${{ always() }}
     steps:
       - name: Indicate completion to coveralls.io
-        run: |
-          pip --no-cache-dir install --upgrade coveralls
-          python -m coveralls --service=github --finish
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        uses: coverallsapp/github-action@v2
+        with:
+          parallel-finished: true

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -47,11 +47,11 @@ jobs:
 
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       # See https://github.com/marketplace/actions/setup-python
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
           cache: 'pip'  # caching pip dependencies

--- a/Dockerfile
+++ b/Dockerfile
@@ -3,11 +3,9 @@ FROM python:3.9-slim-bullseye
 RUN mkdir -p /usr/src/app
 WORKDIR /usr/src/app
 
-COPY requirements.txt /usr/src/app/
-
-RUN pip3 install --no-cache-dir -r requirements.txt
-
 COPY . /usr/src/app
+
+RUN pip3 install --no-cache-dir .
 
 EXPOSE 8080
 

